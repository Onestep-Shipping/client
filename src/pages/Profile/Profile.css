.profile-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  border: solid 2px rgb(0, 153, 180);
  border-radius: 20px;
  padding: 20px;
}

.profile-img {
  width: 250px;
  border-radius: 250px;
  margin-right: 50px;
}

.user-info-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.user-info-row {
  width: 100%;
  margin-bottom: 30px;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.info-label-link {
  font-size: 16px;
  color: rgb(0, 153, 180);
}

#email {
  margin-right: 40px;
}

.tracking-container {
  margin-top: 100px;
  width: 80%;
  padding: 30px 0px;
}

.schedule-result-text-link {
  color: #224DAA;
  cursor: pointer;
  text-decoration: underline;
  font-weight: 700;
}

