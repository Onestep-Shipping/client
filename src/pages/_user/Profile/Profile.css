.profile-container {
  width: 90%;
  padding: 30px 0px;
}

.schedule-result-text-link {
  color: rgb(224, 69, 69);
  font-weight: 600;
  text-decoration: underline;
  cursor: pointer;
}

#red-link {
  color: rgb(38, 126, 11) !important;
  cursor: pointer;
  text-decoration: underline;
  font-weight: 600;
}

.booking-no-button {
  color: #224DAA;
  cursor: pointer;
  text-decoration: underline;
  font-weight: 600;
}

.booking-profile-row {
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  border-radius: 10px;
  padding: 30px 0px;
  user-select: none;
  border: 0.5px dashed rgb(100, 100, 100);
  margin-bottom: 15px;
}

.tooltip-button {
  width: 50%;
  color: #224DAA;
  padding: 5px;
  background-color: white;
  font-size: 14px;
  border: none;
  outline: none;
  cursor: pointer;
}

.tiptool-container {
  width: 150px;
}


