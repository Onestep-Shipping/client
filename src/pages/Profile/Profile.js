import React from 'react';
import './Profile.css';
import {useHistory} from 'react-router-dom';

import Header from '../../components/Header/Header.js';
import DATA from '../../data/ScheduleDetailsData.js';

const Profile = () => {
  const history = useHistory();
  const PROFILE_HEADERS = ['Date Booked', 'From', 'To', 'Vessel', 'Booking Status', 'BOL Status'];

  const chooseBackgrounColorFromStatus = (booking, bol) => {
    if (booking === "In Process" && bol === "In Process") {
      return "#e6e6e6"; // gray
    } else if (booking === "Received" && bol === "Received") {
      return "#a0eec1"; // green
    } else {
      return "#f2e89d"; // yellow
    }
  };

  const handleBook = (status, id) => {
    if (status === "Received") {
      history.push('/bill-of-lading/' + id);
    }
  }

  const handleBol = (status, id) => {
    if (status === "Received") {
      console.log(id);
    }
  }

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <div className="profile-container">
          <div className="schedule-result-header-row">
            {
              PROFILE_HEADERS.map((header, ind) => {
                return (
                  <div className="col" key={ind}>
                    <text className="schedule-result-header-text">{header}</text>
                  </div>
                );
              })
            }
          </div>
          {
            DATA.map((booking, ind) => {
              return (
                <div className="schedule-result-row-container" key={ind}>
                  <div 
                    className='schedule-result-row' 
                    style={{
                      backgroundColor: chooseBackgrounColorFromStatus(booking.bookingStatus, booking.bolStatus)
                    }} >
                    <div className="col">
                      <text className="schedule-result-text">{booking.bookedDate}</text>
                    </div>
                    <div className="col">
                      <text className="schedule-result-text">{booking.from}</text>
                      <text className="schedule-result-text-time">{booking.fromDate}</text>
                    </div>
                    <div className="col">
                      <text className="schedule-result-text">{booking.to}</text>
                      <text className="schedule-result-text-time">{booking.toDate}</text>
                    </div>
                    <div className="col">
                      <text className="schedule-result-text">{booking.ves.substring(0, booking.ves.indexOf('/'))}</text>
                    </div>
                    <div className="col" onClick={() => handleBook(booking.bookingStatus, ind)}>
                      <text 
                        className={"schedule-result-text" + (booking.bookingStatus === "Received" ? "-link" : "")}>
                          {booking.bookingStatus}
                      </text>
                    </div>
                    <div className="col" onClick={() => handleBol(booking.bolStatus, ind)}>
                      <text 
                        className={"schedule-result-text" + (booking.bolStatus === "Received" ? "-link" : "")}>
                          {booking.bolStatus}
                      </text>
                    </div>
                  </div>
                </div>
              );
            })
          }
        </div>
      </div>
    </div>
  );
};

export default Profile;
