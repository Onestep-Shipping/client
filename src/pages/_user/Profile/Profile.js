import React, {useCallback} from 'react';
import './Profile.css';
import { useHistory } from 'react-router-dom';

import Header from '../../../components/Header/Header.js';
import FixedSizeList from '../../../components/FixedSizeList/FixedSizeList.js';
import DATA from '../../../data/ScheduleDetailsData.js';

const Profile = () => {
  const history = useHistory();
  const PROFILE_HEADERS = ['#', 'Date Booked', 'From', 'To', 'Vessel', 'Booking Status', 'BOL Status'];

  const chooseBackgrounColorFromStatus = (booking, bol) => {
    if (booking === "In Process" && bol === "Not Ready") {
      return "#e6e6e6"; // gray
    } else if (booking === "Completed" && bol === "Completed") {
      return "#a0eec1"; // green
    } else {
      return "#f2e89d"; // yellow
    }
  };

  const handleBook = useCallback((status, id) => {
    if (status === "Received") {
      history.push('/bill-of-lading/' + id);
    }
  }, [history]);

  const handleBol = (status, id) => {
    if (status === "Received") {
      console.log(id);
    }
  }

  const row = (booking, ind) => {
    return (
      <div key={ind}>
        <div 
          className='booking-profile-row' 
          style={{
            backgroundColor: chooseBackgrounColorFromStatus(booking.bookingStatus, booking.bolStatus)
          }} >
          <div className="col-numb">
            <text className="schedule-result-text">{ind + 1}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{booking.bookedDate}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{booking.from}</text>
            <text className="schedule-result-text-time">{booking.fromDate}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{booking.to}</text>
            <text className="schedule-result-text-time">{booking.toDate}</text>
          </div>
          <div className="col">
            <text className="schedule-result-text">{booking.ves.substring(0, booking.ves.indexOf('/'))}</text>
          </div>
          <div className="col" onClick={() => handleBook(booking.bookingStatus, ind)}>
            <text
              id={booking.bookingStatus === "Received" ? "red-link" : ""}
              className={"schedule-result-text" + 
                (booking.bookingStatus === "Completed" ? "-link" : "")}>
                {booking.bookingStatus}
            </text>
          </div>
          <div className="col" onClick={() => handleBol(booking.bolStatus, ind)}>
            <text 
              id={booking.bolStatus === "Received" ? "red-link" : ""}
              className={"schedule-result-text" + 
                (booking.bolStatus === "Completed" ? "-link" : "")}>
                {booking.bolStatus}
            </text>
          </div>
        </div>
      </div>
    );
  };

  return (
    <div className="homepage-container">
      <Header />
      <div className="body-container2">
        <div className="profile-container">
          <FixedSizeList headers={PROFILE_HEADERS} data={DATA} row={row}/>
        </div>
      </div>
    </div>
  );
};

export default Profile;
