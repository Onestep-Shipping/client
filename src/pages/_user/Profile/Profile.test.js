import { ApolloProvider } from '@apollo/react-hooks';
import Profile from './Profile';
import React from 'react';
import { shallow } from 'enzyme';

let wrapper = shallow(
  <ApolloProvider>
    <Profile />
  </ApolloProvider>
);

describe('Profile', () => {
  it('should render the Profile Component correctly', () => {   
    expect(Profile).toMatchSnapshot();
  });
});