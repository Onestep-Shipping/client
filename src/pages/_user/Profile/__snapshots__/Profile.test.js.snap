// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render the Profile Component correctly 1`] = `[Function]`;
